### ISSUE TITLE
Graduate TaintNodesByCondition to GA

### ISSUE THREAD

**Issue by ahg-g** [kubernetes/kubernetes#82635]
This feature has been in Beta since 1.12, we can now graduate it to GA.

/sig scheduling
/cc @k82cn

**Comment by ahg-g** [comment]
Note that predicates ```CheckNodeMemoryPressurePred, CheckNodeDiskPressurePred, CheckNodePIDPressurePred, CheckNodeConditionPred``` has been disabled since this feature graduated to beta, i.e., for the past five releases those predicates where practically deprecated and no one would've been able to use them via the Policy API.

Once #82703 is merged, I suggest we remove them completely even though that will require changing the [Policy v1 compatibility test](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/api/compatibility/compatibility_test.go). 

Which brings us to another issue, the compatibility test is not applying the feature gates, which we may be breaking the Policy API compatibility in [ApplyFeatureGates](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithmprovider/defaults/defaults.go#L63) without the test capturing it as is the case with the predicates mentioned above.

/cc @Huang-Wei @alculquicondor

**Comment by draveness** [comment]
ref: https://github.com/kubernetes/enhancements/issues/382
